llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1435,6 +1439,51 @@ bool VPlanTransforms::tryAddExplicitVectorLength(VPlan &Plan) {
   return true;
 }
 
+void VPlanTransforms::addExplicitVectorLengthForCSA(
+    VPInstruction *VPEVL, const MapVector<PHINode *, VPCSAState *> &CSAStates,
+    bool PlanHasScalarVFOnly) {
+
+  // We build the scalar version of a CSA when VF=ElementCount::getFixed(1),
+  // which does not require an EVL.
+  if (PlanHasScalarVFOnly)
+    return;
```

Comment:
Is this a planned extension? I'm not sure what you can do with VF=1, so maybe drop `PlanHasScalarVFOnly` argument and handle in caller?